phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/control/ApplicationSidebarToggleGroup.java

```diff
@@ -0,0 +1,69 @@
+package org.phoenicis.javafx.components.control;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import org.phoenicis.javafx.components.skin.ApplicationSidebarToggleGroupSkin;
+import org.phoenicis.javafx.views.mainwindow.apps.ApplicationsSidebar;
+import org.phoenicis.repository.dto.CategoryDTO;
+
+import java.util.function.Consumer;
+
+/**
+ * A toggle group component used inside the {@link ApplicationsSidebar}
+ */
+public class ApplicationSidebarToggleGroup
+        extends SidebarToggleGroupBase<CategoryDTO, ApplicationSidebarToggleGroup, ApplicationSidebarToggleGroupSkin> {
+    /**
+     * A consumer, which is called when the "all" categories button has been selected
+     */
+    private final ObjectProperty<Runnable> onAllCategorySelection;
```

Comment:
Can't we have this in the base class? Now it's repeated in all derived classes. Same for `onCategorySelection`.

Reply:
Normally yes, but I'm not sure if it should belong to the base class, because it is not used there.
In addition not all `SidebarToggleGroup` implementations require it.
These are the reasons why i decided to add the required callback methods to every `SidebarToggleGroup` implementation.

Reply:
Ok.